src/api/app/helpers/webui/request_helper.rb

```diff
@@ -117,48 +117,66 @@ def diff_label(diff)
 
   # rubocop:disable Style/FormatString
   def request_action_header(action, creator)
-    source_project_hash = { project: action[:sprj], package: action[:spkg], trim_to: nil }
-
-    case action[:type]
+    # FIXME: Remove assignments once we don't call anymore this helper with keys on the right side of the assignment
+    #        (those keys come from bs_request.webui_actions method)
+    action[:source_project] ||= action[:sprj]
+    action[:source_package] ||= action[:spkg]
+    action[:target_project] ||= action[:tprj]
+    action[:target_package] ||= action[:tpkg]
+    action[:target_repository] ||= action[:trepo]
+    action[:person_name] ||= action[:user]
+    action[:group_name] ||= action[:group]
+
+    source_project_hash = { project: action[:source_project], package: action[:source_package], trim_to: nil }
+    target_project_hash = { project: action[:target_project], package: action[:target_package] }
+
+    case action[:type].to_sym
     when :submit
       'Submit %{source_container} to %{target_container}' % {
         source_container: project_or_package_link(source_project_hash),
-        target_container: project_or_package_link(project: action[:tprj], package: action[:tpkg])
+        target_container: project_or_package_link(target_project_hash)
       }
     when :delete
-      target_repository = "repository #{link_to(action[:trepo], repositories_path(project: action[:tprj], repository: action[:trepo]))} for " if action[:trepo]
+      target_repository = if action[:target_repository]
+                            link_to_repository = link_to(action[:target_repository],
+                                                         repositories_path(project: action[:target_project],
+                                                                           repository: action[:target_repository]))
+                            "repository #{link_to_repository} for "
+                          else
+                            ""
```

Comment:
This is triggering a rubocop offense.

```suggestion
                            ''
```